src/dbnode/client/host_queue.go

```diff
@@ -871,13 +871,12 @@ func (q *queue) asyncFetchTagged(op *fetchTaggedOp) {
 	}
 
 	q.Add(1)
-	q.workerPool.GoWithContext(ctx, func() {
-		// NB(r): Defer is slow in the hot path unfortunately
-		cleanup := func() {
-			op.decRef()
-			q.Done()
-		}
-
+	// NB(r): Defer is slow in the hot path unfortunately
+	cleanup := func() {
+		op.decRef()
+		q.Done()
+	}
+	available := q.workerPool.GoWithContext(ctx, func() {
```

Comment:
in reality can this return false?

I think we should find a way that it’s not possible for this to return false (ie it should not block and there should be no way it can return “not available”).

Reply:
Yea good point. This will always spawn a new worker